### PR TITLE
Allow custom error code in throwIf

### LINKED ISSUE #39793: New function: throwError
Is there any way to manually make CH throw a specific error?

Something like this:
```sql
SELECT throwError(999);

Received exception from server (version 22.2.2):
Code: 999. DB::Exception: KEEPER_EXCEPTION ...
```

PS:
*Although there is a similar function (`throwIf`), it only returns code=395 (FUNCTION_THROW_IF_VALUE_IS_NON_ZERO).*

**Use case**

We're looking something like this so we can simulate errors and take relevant actions agains some errors.

### PR DESCRIPTION
**Pull request by rschu1ze**
Fixes #39793

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
It is now possible to set a custom error code for the exception thrown by function "throwIf"

### CODE REVIEW

File: docs/ru/sql-reference/functions/other-functions.md [hunk 1]
```diff
@@ -1727,10 +1727,13 @@ SELECT joinGet(db_test.id_val,'val',toUInt32(number)) from numbers(4) SETTINGS j
 
 Принимает на вход имя и аргументы модели. Возвращает Float64.
 
-## throwIf(x\[, custom_message\]) {#throwifx-custom-message}
+## throwIf(x\[, message\[, error_code\]\]) {#throwifx-custom-message}
 
 Бросает исключение, если аргумент не равен нулю.
-custom_message - необязательный параметр, константная строка, задает текст сообщения об ошибке.
+`custom_message` - необязательный параметр, константная строка, задает текст сообщения об ошибке.
+`error_code` - необязательный параметр, константое число, задает устанавливает код ошибки.
```
Comment (yakov-olkhovskiy) [inline docs/ru/sql-reference/functions/other-functions.md:1734]:
```suggestion
`error_code` - необязательный параметр, константое число, задает код ошибки.
```

**Review by alexey-milovidov — APPROVED** [review]

**Review by yakov-olkhovskiy — APPROVED** [review]

File: src/Functions/throwIf.cpp [hunk 2]
```diff
@@ -22,6 +22,11 @@ namespace ErrorCodes
 namespace
 {
 
+/// The regex-based code style check script in CI complains when it sees "ErrorCodes:: ErrorCode" (space added to avoid another match).
+/// Because this expression is only used in this file, don't add some suppression mechanism to the already complex style checker, instead
+/// work around by creating a namespace alias.
+namespace ErrorCodeAlias = ErrorCodes;
+
```
Comment (yakov-olkhovskiy) [inline src/Functions/throwIf.cpp:29]: I was wondering what da heck style check is complaining about :)

**Review by yakov-olkhovskiy — APPROVED** [review]

File: tests/queries/0_stateless/00602_throw_if.sh [hunk 3]
```diff
@@ -4,9 +4,11 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 # shellcheck source=../shell_config.sh
 . "$CURDIR"/../shell_config.sh
 
-default_exception_message="Value passed to 'throwIf' function is non zero"
+default_exception_message="Value passed to 'throwIf' function is non-zero"
 custom_exception_message="Number equals 1000000"
+custom_error_code="42"
```
Comment (tavplubix) [inline tests/queries/0_stateless/00602_throw_if.sh:9]: Let's also add tests with some special error codes that can be processed special way (like `UNEXPECTED_PACKET_FROM_CLIENT` in `TCPHandler` or `OK`) and error codes that does not exist (like `123456` or `-1`)

File: src/Functions/throwIf.cpp (hunk 2 above)
Comment (rschu1ze) [reply src/Functions/throwIf.cpp:29]: It says that "ErrorCodes::E" is used but not defined :grimacing: ... It's basically a shortcoming of being based on regexes.

File: tests/queries/0_stateless/00602_throw_if.sh (hunk 3 above)
Comment (rschu1ze) [reply tests/queries/0_stateless/00602_throw_if.sh:9]: Sure, will do.

**Comment by rschu1ze** [comment]
Test errors look unrelated.

**Comment by tavplubix** [comment]
AST fuzzer (MSan) - looks like https://github.com/ClickHouse/ClickHouse/issues/40164
Integration tests (asan) [1/3] - test_concurrent_backups_s3/test.py::test_concurrent_backups - @vitlibar, please take a look
Stateless tests (aarch64) - 00993_system_parts_race_condition_drop_zookeeper - https://github.com/ClickHouse/ClickHouse/pull/40246

**Pull request merged**